tests/store/artifact/test_azure_data_lake_artifact_repo.py

```diff
@@ -0,0 +1,229 @@
+    assert parse(global_abfs_without_path2) == ("filesystem", "acct", "")
+
+    global_abfs_with_multi_path = "abfss://filesystem@acct.dfs.core.windows.net/a/b"
+    assert parse(global_abfs_with_multi_path) == ("filesystem", "acct", "a/b")
+
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acct.dfs.core.evil.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acct/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://acct.dfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://@acct.dfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acctxdfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="Not an ABFSS URI"):
+        parse("abfs://cont@acct.dfs.core.windows.net/path")
+
+
+def test_list_artifacts_empty(mock_data_lake_client):
+    repo = AzureDataLakeArtifactRepository(TEST_DATA_LAKE_URI, None)
+    mock_data_lake_client.get_file_system_client.get_paths.return_value = MockPathList([])
+    assert repo.list_artifacts() == []
+
+
+def test_list_artifacts_single_file(mock_data_lake_client):
+    repo = AzureDataLakeArtifactRepository(TEST_DATA_LAKE_URI, mock_data_lake_client)
+
+    # Evaluate single file
+    path_props = PathProperties(name=posixpath.join(TEST_DATA_LAKE_URI, "file"), content_length=42)
+    mock_data_lake_client.get_file_system_client.get_paths.return_value = MockPathList([path_props])
+    assert repo.list_artifacts("file") == []
+
+
+def test_list_artifacts(mock_data_lake_client, mock_filesystem_client):
+    repo = AzureDataLakeArtifactRepository(TEST_DATA_LAKE_URI, None)
+
+    # Create some files to return
+    dir_prefix = PathProperties(is_directory=True, name=posixpath.join(TEST_ROOT_PATH, "dir"))
+    path_props = PathProperties(content_length=42, name=posixpath.join(TEST_ROOT_PATH, "file"))
+    mock_filesystem_client.get_paths.return_value = MockPathList([dir_prefix, path_props])
+
+    artifacts = repo.list_artifacts()
+    mock_filesystem_client.get_paths.assert_called_with(path=TEST_ROOT_PATH, recursive=False)
+    assert artifacts[0].path == "dir"
+    assert artifacts[0].is_dir is True
+    assert artifacts[0].file_size is None
+    assert artifacts[1].path == "file"
+    assert artifacts[1].is_dir is False
+    assert artifacts[1].file_size == 42
+
+    repo.list_artifacts(path="nonexistent-dir")
+    mock_filesystem_client.get_paths.assert_called_with(
+        path=posixpath.join(TEST_ROOT_PATH, "nonexistent-dir"), recursive=False
+    )
+
+
+def test_log_artifacts(
+    mock_data_lake_client, mock_filesystem_client, mock_directory_client, mock_file_client, tmpdir
```

Comment:
```suggestion
    mock_data_lake_client, mock_filesystem_client, mock_directory_client, mock_file_client, tmp_path
```

Can we use `tmp_path`? `tmpdir` is not recommended.

https://py.readthedocs.io/en/latest/path.html#py-path

> The ‘py’ library is in “maintenance mode” and so is not recommended for new projects. Please check out pathlib or pathlib2 for path operations.